vaadin-messages-flow-parent/vaadin-messages-flow/src/main/java/com/vaadin/flow/component/messages/MessageInput.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.messages;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.EventData;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Server-side component for the {@code vaadin-message-input} element. The
+ * component displays a text-area to input a message and a button to send it.
+ *
+ * @author Vaadin Ltd.
+ */
+@Tag("vaadin-message-input")
+@JsModule("@vaadin/vaadin-messages/src/vaadin-message-input.js")
+@NpmPackage(value = "@vaadin/vaadin-messages", version = "v2.0.0-alpha1")
+public class MessageInput extends Component {
```

Comment:
For most components, it makes sense to implement `HasSize` and `HasStyle`, and I think that `MessageInput` is not an exception.